src/libraries/System.Security.Cryptography.Pkcs/tests/Rfc3161/TimestampTokenTests.cs

```diff
@@ -85,10 +85,20 @@ public static void ParseDocument_ExcessData(string testDataName)
 
                 using (var signerCert = new X509Certificate2(testData.ExternalCertificateBytes))
                 {
-                    // Assert.NoThrow
-                    signedCms.CheckSignature(
-                        new X509Certificate2Collection(signerCert),
-                        true);
+                    if (!SignatureSupport.SupportsRsaSha1Signatures && signerCert.SignatureAlgorithm.FriendlyName == "sha512RSA")
+                    {
+                        Assert.ThrowsAny<CryptographicException>(() => signedCms.CheckSignature(
+                                                new X509Certificate2Collection(signerCert),
+                                                true));
```

Comment:
Indentation here is a little off.

```suggestion
                        Assert.ThrowsAny<CryptographicException>(() => signedCms.CheckSignature(
                            new X509Certificate2Collection(signerCert),
                            true));
```